pkg/featuregate/featuregate.go

```diff
@@ -90,6 +90,14 @@ var (
 		featuregate.WithRegisterDescription("enables features associated to the Prometheus Operator"),
 		featuregate.WithRegisterFromVersion("v0.82.0"),
 	)
+
+	// CreateRBACPermissions is the feature gate that enables the automatic creation of RBAC permissions depending on the used processors.
+	CreateRBACPermissions = featuregate.GlobalRegistry().MustRegister(
+		"operator.collector.createrbacpremissions",
+		featuregate.StageAlpha,
+		featuregate.WithRegisterDescription("enables the automatic creation of RBAC permissions depending on the used processors"),
+		featuregate.WithRegisterFromVersion("v0.91.0"),
+	)
```

Comment:
I'm wondering if we shouldn't make this an operator-level configuration option instead of a feature flag. A feature flag implies we eventually want to make this behavior impossible to disable, and I can imagine users wanting to disable it and handle RBAC on their own.

Reply:
Right now I see this as a way to enable/disable the feature. Not as something to be "mandatory" in the future. If this is not the correct way of doing it, I can change it. Would it be a better idea to have a CLI argument or how do you propose to implement this?

Reply:
Did we explore to make it work without the feature flag? The operator could issue SAR request to figure out its permissions and act accordingly or just log warning (send warning in webhook) if it cannot create RBAC.